src/tracker_manager.cpp

```diff
@@ -241,16 +241,16 @@ namespace libtorrent
 	void tracker_manager::remove_request(tracker_connection const* c)
 	{
 		TORRENT_ASSERT(is_single_thread());
-		http_conns_t::iterator i = std::find_if(m_http_conns.begin(), m_http_conns.end()
+		auto const i = std::find_if(m_http_conns.begin(), m_http_conns.end()
 			, [c] (std::shared_ptr<http_tracker_connection> const& ptr) { return ptr.get() == c; });
 		if (i != m_http_conns.end())
 		{
 			m_http_conns.erase(i);
 			return;
 		}
 
-		udp_conns_t::iterator j = std::find_if(m_udp_conns.begin(), m_udp_conns.end()
-			, [c] (udp_conns_t::value_type const& uc) { return uc.second.get() == c; });
+		auto const j = std::find_if(m_udp_conns.begin(), m_udp_conns.end()
+			, [c] (decltype(m_udp_conns)::value_type const& uc) { return uc.second.get() == c; });
```

Comment:
I will use the corresponding type, I don't like to have necessary `typedef`s. You tell me if it's ok.

Any reason against the use of`decltype`?